### PR TITLE
refactor!: hide UI implementation details

### PR DESCRIPTION
**Pull request by redeboer**
Addresses #24, but doesn't fully close it, because there are many other modules that still need to have their implementation details hidden. Not that hiding implementation details often causes issues in the API for `intersphinx`.

*Breaking change*: the `default_settings` and `system_control` modules have become completely hidden. Therefore, the `StateTransitionManager` has been moved to `ui`.

### CODE REVIEW

**Comment by redeboer** [comment]
@spflueger Could you check if some of the methods of the `StateTransisionManager` could be made private? That would simplify the API further.

File: expertsystem/ui/__init__.py [hunk 1]
```diff
@@ -1,9 +1,358 @@
...
+        topology_graphs = self.build_topologies()
+        init_graphs = self.create_seed_graphs(topology_graphs)
+        graph_node_setting_pairs = self.determine_node_settings(init_graphs)
+        # create groups of settings ordered by "probability"
+        graph_settings_groups = create_interaction_setting_groups(
+            graph_node_setting_pairs
+        )
+        return graph_settings_groups
+
+    def build_topologies(self):
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:161]: this can be made private

File: expertsystem/ui/__init__.py [hunk 2]
```diff
@@ -1,9 +1,358 @@
...
+        all_graphs = self.topology_builder.build_graphs(
+            len(self.initial_state), len(self.final_state)
+        )
+        logging.info(f"number of topology graphs: {len(all_graphs)}")
+        return all_graphs
+
+    def create_seed_graphs(self, topology_graphs):
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:168]: this too

File: expertsystem/ui/__init__.py [hunk 3]
```diff
@@ -1,9 +1,358 @@
...
+                    self.initial_state,
+                    self.final_state,
+                    self.final_state_groupings,
+                )
+            )
+
+        logging.info(f"initialized {len(init_graphs)} graphs!")
+        return init_graphs
+
+    def determine_node_settings(self, graphs):
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:187]: this too

File: expertsystem/ui/__init__.py [hunk 4]
```diff
@@ -1,9 +1,358 @@
...
+        match_external_edges(solutions)
+        final_solutions = []
+        for sol in solutions:
+            final_solutions.extend(
+                perform_external_edge_identical_particle_combinatorics(sol)
+            )
+
+        return (final_solutions, violated_laws)
+
+    def propagate_quantum_numbers(self, state_graph_node_settings_pair):
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:320]: this too

File: expertsystem/ui/__init__.py [hunk 5]
```diff
@@ -1,9 +1,358 @@
...
+        propagator = self.initialize_qn_propagator(
+            state_graph_node_settings_pair[0],
+            state_graph_node_settings_pair[1],
+        )
+        solutions = propagator.find_solutions()
+        return (solutions, propagator.get_non_satisfied_conservation_laws())
+
+    def initialize_qn_propagator(self, state_graph, node_settings):
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:328]: this too

Comment (redeboer) [reply expertsystem/ui/__init__.py]: Ok perfecto, thanks

**Pull request merged**